### ISSUE TITLE
Failure on qt-attr

### ISSUE THREAD

**Issue by genelkim** [genelkim/ulf-sanity-checker#30]
************************************
## Sanity checking formula (before preprocessing).
```
(|"|
 (I.PRO
  ((PRES WILL.AUX-S)
   (SHOW.V YOU.PRO (THE.D (N+PREDS WAY.N (BACK.P REF.PRO)))
    (ADV-A (BY.P MYSELF.PRO)))))
 (QT-ATTR ((THE.D CAPTAIN.N) ((PAST SAY.V) *QT)))
 (THAT.PRO
  ((PRES BE.V)
   (= (THE.D
       (N+PREDS (LEAST.MOD-N HONOR.N)
        (SUB THAT.REL
             (I.PRO
              ((PRES CAN.AUX-V)
               (RENDER.V *H (TO.P-ARG (YOUR.D EXCELLENCY.N)))))))))))
 |"|)
```

## Sanity checking formula (after preprocessing).
```
((|"|
  (I.PRO
   ((PRES WILL.AUX-S)
    ((ADV-A (BY.P MYSELF.PRO))
     (SHOW.V YOU.PRO (THE.D (N+PREDS WAY.N (BACK.P REF.PRO)))))))
  (QT-ATTR ((THE.D CAPTAIN.N) ((PAST SAY.V) *QT)))
  (THAT.PRO
   ((PRES BE.V)
    (= (THE.D
        (N+PREDS (LEAST.MOD-N HONOR.N)
         (I.PRO
          ((PRES CAN.AUX-V)
           (RENDER.V THAT.REL (TO.P-ARG (YOUR.D EXCELLENCY.N))))))))))
  |"|)
 NIL NIL)
```

## Possible errors

Possibly failed conditions:
  _"UNKNOWN detected in type analysis.  Please ensure this isn't from an annotation error."_
Ann segment:
  ```
NIL
```
Predicted constituent types ((list of types) -- constituent)

## Formula with predicted types (for debugging).

```
(((TYPES TERM)
  (|"|
   ((TYPES TENSED-SENT)
    (I.PRO
     ((TYPES PRED TENSED-VERB)
      (((TYPES TENSED-AUX) (PRES WILL.AUX-S))
       ((TYPES VERB PRED)
        (((TYPES ADV-A) (ADV-A ((TYPES PP PRED) (BY.P MYSELF.PRO))))
         ((TYPES VERB PRED)
          (SHOW.V YOU.PRO
           ((TYPES TERM)
            (THE.D
             ((TYPES NOUN PRED)
              (N+PREDS WAY.N
               ((TYPES PP PRED) (BACK.P REF.PRO))))))))))))))
   ((TYPES UNKNOWN)
    (QT-ATTR
     ((TYPES TENSED-SENT)
      (((TYPES TERM) (THE.D CAPTAIN.N))
       ((TYPES PRED TENSED-VERB)
        (((TYPES PRED TENSED-VERB) (PAST SAY.V)) *QT))))))
   ((TYPES PRED TENSED-SENT REL-SENT)
    (THAT.PRO
     ((TYPES PRED TENSED-VERB)
      (((TYPES PRED TENSED-VERB) (PRES BE.V))
       ((TYPES ADJ PRED)
        (= ((TYPES TERM)
            (THE.D
             ((TYPES NOUN PRED)
              (N+PREDS ((TYPES NOUN PRED) (LEAST.MOD-N HONOR.N))
               ((TYPES PRED TENSED-SENT REL-SENT)
                (I.PRO
                 ((TYPES PRED TENSED-VERB)
                  (((TYPES TENSED-AUX) (PRES CAN.AUX-V))
                   ((TYPES VERB PRED)
                    (RENDER.V THAT.REL
                     ((TYPES P-ARG)
                      (TO.P-ARG
                       ((TYPES TERM)
                        (YOUR.D EXCELLENCY.N))))))))))))))))))))
   |"|)))
```

NIL

####################################

**Comment by genelkim** [comment]
Another example 943484

************************************
## Sanity checking formula (before preprocessing).
```
(|"|
 (AND.ADV-S NOW.ADV-E (VOC (YOUR.D EXCELLENCY.N))
  (QT-ATTR (HE.PRO ((PAST ADD.V) *QT)))
  (((YOU.PRO
     ((PRES ALLOW.V) ME.PRO (TO (REPEAT.V (MY.D (PLUR APOLOGY.N))))))
    !)
   AND.CC
   (I.PRO
    ((PRES HOPE.V)
     (THT
      (YOU.PRO
       ((PRES WILL.AUX-S) NOT
        (ENTERTAIN.V
         (ANY.D
          (RESENTMENT.N
           (AT.P-ARG (NP+PREDS WHAT.PRO (PERF OCCUR.V)))))))))))))
 |"|)
```

## Sanity checking formula (after preprocessing).
```
((|"|
  ((QT-ATTR (HE.PRO ((PAST ADD.V) *QT)))
   (((YOU.PRO
      ((PRES ALLOW.V) ME.PRO (TO (REPEAT.V (MY.D (PLUR APOLOGY.N))))))
     !)
    AND.CC
    (I.PRO
     ((PRES HOPE.V)
      (THT
       (YOU.PRO
        ((PRES WILL.AUX-S)
         (ENTERTAIN.V
          (ANY.D
           (RESENTMENT.N
            (AT.P-ARG (NP+PREDS WHAT.PRO (PERF OCCUR.V)))))))))))))
  |"|)
 (AND.ADV-S NOW.ADV-E NOT) ((VOC (YOUR.D EXCELLENCY.N))))
```

## Possible errors

Possibly failed conditions:
  _"UNKNOWN detected in type analysis.  Please ensure this isn't from an annotation error."_
Ann segment:
  ```
NIL
```
Predicted constituent types ((list of types) -- constituent)

## Formula with predicted types (for debugging).

```
(((TYPES TERM)
  (|"|
   ((TYPES UNKNOWN)
    (((TYPES UNKNOWN)
      (QT-ATTR
       ((TYPES TENSED-SENT)
        (HE.PRO
         ((TYPES PRED TENSED-VERB)
          (((TYPES PRED TENSED-VERB) (PAST ADD.V)) *QT))))))
     ((TYPES TENSED-SENT)
      (((TYPES TENSED-SENT)
        (((TYPES TENSED-SENT)
          (YOU.PRO
           ((TYPES PRED TENSED-VERB)
            (((TYPES PRED TENSED-VERB) (PRES ALLOW.V)) ME.PRO
             ((TYPES TERM)
              (TO
               ((TYPES VERB PRED)
                (REPEAT.V
                 ((TYPES TERM)
                  (MY.D ((TYPES NOUN PRED) (PLUR APOLOGY.N))))))))))))
         !))
       AND.CC
       ((TYPES TENSED-SENT)
        (I.PRO
         ((TYPES PRED TENSED-VERB)
          (((TYPES PRED TENSED-VERB) (PRES HOPE.V))
           ((TYPES TERM)
            (THT
             ((TYPES TENSED-SENT)
              (YOU.PRO
               ((TYPES PRED TENSED-VERB)
                (((TYPES TENSED-AUX) (PRES WILL.AUX-S))
                 ((TYPES VERB PRED)
                  (ENTERTAIN.V
                   ((TYPES TERM)
                    (ANY.D
                     ((TYPES NOUN PRED)
                      (RESENTMENT.N
                       ((TYPES P-ARG)
                        (AT.P-ARG
                         ((TYPES TERM)
                          (NP+PREDS WHAT.PRO
                           ((TYPES VERB PRED)
                            (PERF OCCUR.V))))))))))))))))))))))))))
   |"|))
 AND.ADV-S NOW.ADV-E NOT
 ((TYPES UNKNOWN) (VOC ((TYPES TERM) (YOUR.D EXCELLENCY.N)))))
```

NIL

####################################

**Comment by genelkim** [comment]
943487
************************************
## Sanity checking formula (before preprocessing).
```
(|"|
 (((VOC (K (PLUR GENTLEMAN.N)))
   (QT-ATTR
    ((THE.D CHIEF.N) (PAST ADD.V) *H
     (ADV-A
      (TURN.V (ADV-A (TOWARDS.P (THE.D (YOUNG.A (PLUR MAN.N)))))))))
   (PERHAPS.ADV-S
    ((THE.D OFFER.N)
     ((PRES MAY.AUX-S) NOT
      (APPEAR.V (VERY.MOD-A TEMPTING.A) (TO.P-ARG YOU.PRO))))))
  (BUT.ADV-S
   ((IF.PS
     (YOU.PRO
      ((CF SHOULD.AUX-S) EVER.ADV-E
       (FEEL.V
        (INCLINED.A
         (TO
          (PAY.V ME.PRO (A.D (SECOND.A VISIT.N))
           (ADV-E
            (AT.P
             (ANY.D
              (N+PREDS PLACE.N
               (SUB WHERE.REL
                    (I.PRO
                     ((CF MAY.AUX-S) (BE.V (AT.P *H))))))))))))))))
    (YOU.PRO ((PRES SHALL.AUX-S) (BE.V WELCOME.A))))))
 |"|)
```

## Sanity checking formula (after preprocessing).
```
((|"|
  (((QT-ATTR
     ((THE.D CHIEF.N) (PAST ADD.V) *H
      (ADV-A
       (TURN.V (ADV-A (TOWARDS.P (THE.D (YOUNG.A (PLUR MAN.N)))))))))
    (PERHAPS.ADV-S
     ((THE.D OFFER.N)
      ((PRES MAY.AUX-S)
       (APPEAR.V (VERY.MOD-A TEMPTING.A) (TO.P-ARG YOU.PRO))))))
   (BUT.ADV-S
    ((IF.PS
      (YOU.PRO
       ((CF SHOULD.AUX-S)
        (FEEL.V
         (INCLINED.A (TO (PAY.V ME.PRO (A.D (SECOND.A VISIT.N)))))))))
     (YOU.PRO ((PRES SHALL.AUX-S) (BE.V WELCOME.A))))))
  |"|)
 (NOT EVER.ADV-E
      (ADV-E
       (AT.P
        (ANY.D
         (N+PREDS PLACE.N
          (I.PRO ((CF MAY.AUX-S) (BE.V (AT.P WHERE.REL)))))))))
 ((VOC (K (PLUR GENTLEMAN.N)))))
```

## Possible errors

Possibly failed conditions:
  _"'qt-attr' operator should take one argument and it should contain a '*qt'"_
Ann segment:
  ```
(QT-ATTR
 ((THE.D CHIEF.N) (PAST ADD.V) [*H]
  (ADV-A (TURN.V (ADV-A (TOWARDS.P (THE.D (YOUNG.A (PLUR MAN.N)))))))))
```
Predicted constituent types ((list of types) -- constituent)
  ```
((MACRO-SYMBOL) -- QT-ATTR)
```
  ```
((UNKNOWN) --
 ((THE.D CHIEF.N) (PAST ADD.V) [*H]
  (ADV-A (TURN.V (ADV-A (TOWARDS.P (THE.D (YOUNG.A (PLUR MAN.N)))))))))
```

Possibly failed conditions:
  _"UNKNOWN detected in type analysis.  Please ensure this isn't from an annotation error."_
Ann segment:
  ```
NIL
```
Predicted constituent types ((list of types) -- constituent)

## Formula with predicted types (for debugging).

```
(((TYPES TERM)
  (|"|
   ((TYPES UNKNOWN)
    (((TYPES UNKNOWN)
      (((TYPES UNKNOWN)
        (QT-ATTR
         ((TYPES UNKNOWN)
          (((TYPES TERM) (THE.D CHIEF.N))
           ((TYPES PRED TENSED-VERB) (PAST ADD.V)) *H
           ((TYPES ADV-A)
            (ADV-A
             ((TYPES VERB PRED)
              (TURN.V
               ((TYPES ADV-A)
                (ADV-A
                 ((TYPES PP PRED)
                  (TOWARDS.P
                   ((TYPES TERM)
                    (THE.D
                     ((TYPES NOUN PRED)
                      (YOUNG.A
                       ((TYPES NOUN PRED) (PLUR MAN.N))))))))))))))))))
       ((TYPES TENSED-SENT)
        (PERHAPS.ADV-S
         ((TYPES TENSED-SENT)
          (((TYPES TERM) (THE.D OFFER.N))
           ((TYPES PRED TENSED-VERB)
            (((TYPES TENSED-AUX) (PRES MAY.AUX-S))
             ((TYPES VERB PRED)
              (APPEAR.V ((TYPES ADJ PRED) (VERY.MOD-A TEMPTING.A))
               ((TYPES P-ARG) (TO.P-ARG YOU.PRO))))))))))))
     ((TYPES TENSED-SENT)
      (BUT.ADV-S
       ((TYPES TENSED-SENT)
        (((TYPES SENT-MOD)
          (IF.PS
           ((TYPES TENSED-SENT)
            (YOU.PRO
             ((TYPES PRED TENSED-VERB)
              (((TYPES TENSED-AUX) (CF SHOULD.AUX-S))
               ((TYPES VERB PRED)
                (FEEL.V
                 ((TYPES ADJ PRED)
                  (INCLINED.A
                   ((TYPES TERM)
                    (TO
                     ((TYPES VERB PRED)
                      (PAY.V ME.PRO
                       ((TYPES TERM)
                        (A.D
                         ((TYPES NOUN PRED)
                          (SECOND.A VISIT.N))))))))))))))))))
         ((TYPES TENSED-SENT)
          (YOU.PRO
           ((TYPES PRED TENSED-VERB)
            (((TYPES TENSED-AUX) (PRES SHALL.AUX-S))
             ((TYPES VERB PRED) (BE.V WELCOME.A))))))))))))
   |"|))
 NOT EVER.ADV-E
 ((TYPES ADV-E SENT-MOD)
  (ADV-E
   ((TYPES PP PRED)
    (AT.P
     ((TYPES TERM)
      (ANY.D
       ((TYPES NOUN PRED)
        (N+PREDS PLACE.N
         ((TYPES PRED TENSED-SENT REL-SENT)
          (I.PRO
           ((TYPES PRED TENSED-VERB)
            (((TYPES TENSED-AUX) (CF MAY.AUX-S))
             ((TYPES VERB PRED)
              (BE.V ((TYPES PP PRED) (AT.P WHERE.REL))))))))))))))))
 ((TYPES UNKNOWN)
  (VOC ((TYPES TERM) (K ((TYPES NOUN PRED) (PLUR GENTLEMAN.N)))))))
```

NIL

####################################

**Comment by genelkim** [comment]
943492
************************************
## Sanity checking formula (before preprocessing).
```
(|"|
 ((YES.YN (I.PRO ((PRES HAVE.V) REF.PRO)))
  (QT-ATTR (|Franz| ((PAST REPLY.V) *H)))
  (I.PRO
   ((PRES BE.V)
    (CURIOUS.A
     (TO
      (KNOW.V
       (ANS-TO
        (SUB (WHAT.D WORK.N)
             (YOU.PRO
              ((PAST PROG)
               (PERUSE.V *H
                (ADV-A (WITH.P ((SO.MOD-A MUCH.A) ATTENTION.N)))
                (ADV-E (AS.P (WE.PRO (PAST ENTER.V)))))))))))))))
 |"|)
```

## Sanity checking formula (after preprocessing).
```
((|"|
  ((YES.YN (I.PRO ((PRES HAVE.V) REF.PRO)))
   (QT-ATTR (|Franz| ((PAST REPLY.V) *H)))
   (I.PRO
    ((PRES BE.V)
     (CURIOUS.A
      (TO
       (KNOW.V
        (ANS-TO
         (YOU.PRO
          ((PAST PROG)
           ((ADV-A (WITH.P ((SO.MOD-A MUCH.A) ATTENTION.N)))
            (PERUSE.V (WHAT.D WORK.N))))))))))))
  |"|)
 ((ADV-E (AS.P (WE.PRO (PAST ENTER.V))))) NIL)
```

## Possible errors

Possibly failed conditions:
  _"'qt-attr' operator should take one argument and it should contain a '*qt'"_
Ann segment:
  ```
(QT-ATTR (|Franz| ((PAST REPLY.V) [*H])))
```
Predicted constituent types ((list of types) -- constituent)
  ```
((MACRO-SYMBOL) -- QT-ATTR)
```
  ```
((TENSED-SENT) -- (|Franz| ((PAST REPLY.V) [*H])))
```

Possibly failed conditions:
  _"Simple prepositions (*.p) take terms as arguments and take 1 argument."_
Ann segment:
  ```
(WITH.P ((SO.MOD-A MUCH.A) ATTENTION.N))
```
Predicted constituent types ((list of types) -- constituent)
  ```
((UNKNOWN) -- WITH.P)
```
  ```
((NOUN PRED) -- ((SO.MOD-A MUCH.A) ATTENTION.N))
```

Possibly failed conditions:
  _"Simple prepositions (*.p) take terms as arguments and take 1 argument."_
Ann segment:
  ```
(AS.P (WE.PRO (PAST ENTER.V)))
```
Predicted constituent types ((list of types) -- constituent)
  ```
((UNKNOWN) -- AS.P)
```
  ```
((TENSED-SENT) -- (WE.PRO (PAST ENTER.V)))
```

Possibly failed conditions:
  _"UNKNOWN detected in type analysis.  Please ensure this isn't from an annotation error."_
Ann segment:
  ```
NIL
```
Predicted constituent types ((list of types) -- constituent)

## Formula with predicted types (for debugging).

```
(((TYPES TERM)
  (|"|
   ((TYPES UNKNOWN)
    (((TYPES TENSED-SENT)
      (YES.YN
       ((TYPES TENSED-SENT)
        (I.PRO
         ((TYPES PRED TENSED-VERB)
          (((TYPES PRED TENSED-VERB) (PRES HAVE.V)) REF.PRO))))))
     ((TYPES UNKNOWN)
      (QT-ATTR
       ((TYPES TENSED-SENT)
        (|Franz|
         ((TYPES PRED TENSED-VERB)
          (((TYPES PRED TENSED-VERB) (PAST REPLY.V)) *H))))))
     ((TYPES TENSED-SENT)
      (I.PRO
       ((TYPES PRED TENSED-VERB)
        (((TYPES PRED TENSED-VERB) (PRES BE.V))
         ((TYPES ADJ PRED)
          (CURIOUS.A
           ((TYPES TERM)
            (TO
             ((TYPES VERB PRED)
              (KNOW.V
               ((TYPES TERM)
                (ANS-TO
                 ((TYPES TENSED-SENT)
                  (YOU.PRO
                   ((TYPES PRED TENSED-VERB)
                    (((TYPES TENSED-AUX) (PAST PROG))
                     ((TYPES VERB PRED)
                      (((TYPES ADV-A)
                        (ADV-A
                         ((TYPES PP PRED)
                          (WITH.P
                           ((TYPES NOUN PRED)
                            (((TYPES ADJ PRED) (SO.MOD-A MUCH.A))
                             ATTENTION.N))))))
                       ((TYPES VERB PRED)
                        (PERUSE.V
                         ((TYPES TERM)
                          (WHAT.D WORK.N))))))))))))))))))))))))
   |"|))
 ((TYPES ADV-E SENT-MOD)
  (ADV-E
   ((TYPES PP PRED)
    (AS.P
     ((TYPES TENSED-SENT)
      (WE.PRO ((TYPES PRED TENSED-VERB) (PAST ENTER.V)))))))))
```

NIL

####################################

**Comment by genelkim** [comment]
Fixed in db1cc1155d47ce92679f321e61d5ff6e9e5b1ee9 -- wait for update on production system
